docs/QuickReference.html

```diff
@@ -2,92 +2,216 @@
-<p>class MyMultiset<t> { /*…*/ }datatype Tree</t><t> = Leaf | Node(Tree</t><t>, T, Tree</t><t>)method Find</t><t>(key: T, collection: Tree</t><t>) { /*…*/ }function IfThenElse</t><t>(b: bool, x: T, y: T): T { /*…*/ }</t></p>
+<p>Dafny supports type parameters.  That is, any class, inductive datatype, method, and function can have type parameters. These are declared in angle brackets after the name of what is being declared. For example:</p>
+<pre>
+class MyMultiset<t> {
+  /*...*/ 
+}
+
+datatype Tree</t><t> = Leaf | Node(Tree</t><t>, T, Tree</t><t>)
+
+method Find</t><t>(key: T, collection: Tree</t><t>) { 
+  /*...*/ 
+}
+
+function IfThenElse</t><t>(b: bool, x: T, y: T): T { 
+  /*...*/ 
+}</pre>
 <h2>Statements</h2>
 <p>Here are examples of the most common statements in Dafny.</p>
-<p>var LocalVariables := ExprList;Lvalues := ExprList;assert BoolExpr;print ExprList;</p>
-<p>if BoolExpr0 {Stmts0} else if BoolExpr1 {Stmts1} else {Stmts2}</p>
-<p>while BoolExprinvariant Invmodifies Framedecreases Rank{Stmts}</p>
-<p>match Expr {case Empty => Stmts0case Node(l, d, r) => Stmts1}</p>
-<p>break;return;</p>
+<pre>
+  var LocalVariables := ExprList;
+
+  Lvalues := ExprList;
+
+  assert BoolExpr;print ExprList;
+
+  if BoolExpr0 {
+    Stmts0
+  } else if BoolExpr1 {
+    Stmts1
+  } else {
+    Stmts2
+  }
+
+  while BoolExpr 
+    invariant Inv
+    modifies Frame
+    decreases Rank
+  {
+    Stmts
+  }
+
+  match Expr {
+    case Empty => Stmts0
+    case Node(l, d, r) => Stmts1
+  }
+
+  break;
+
+  return;
+</pre>
 <p>The var statement introduces local variables (which are not allowed to shadow other variables declared inside the same set of most tightly enclosing curly braces). Each variable can optionally be followed by :T for any type T, which explicitly gives the preceding variable the type T (rather than being inferred). The ExprList with initial values is optional. To declare the variables as ghost variables, precede the declaration with the keyword ghost.</p>
 <p>The assignment statement assigns each right-hand side in ExprList to the corresponding left-hand side in Lvalues. These assignments are performed in parallel (more to the point, all necessary reads occur before the writes), so the left-hand sides must denote distinct L-values. Each right-hand side can be an expression or an object creation of one of the following forms:</p>
-<p>new Tnew T.Init(ExprList)new T(ExprList)new T[SizeExpr]new T[SizeExpr0, SizeExpr1]</p>
-<p>The first form allocates an object of type T. The second form additionally invokes an initialization method or constructor on the newly allocated object. The third form shows how the syntax differs from the second form when the constructor called is anonymous.  The other forms show examples of array allocations, in particular a one- and a two-dimensional array of T values, respectively.</p>
+<pre>
+  new Tnew T.Init(ExprList)
```

Comment:
break into two lines

Reply:
Add an example that shows the syntax when the constructor is anonymous:

    new T(ExprList)

Reply:
Done